### PR TITLE
Company edit refactor

### PR DESCRIPTION
**Pull request by teneightfive**
This work tries to remove more duplication from the controllers and middleware used to add and edit companies.

This work is still not complete and needs more refinement to bring the view in line with newer form macros.

### CODE REVIEW

File: src/apps/companies/middleware/params.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+const { getDitCompany, getCHCompany } = require('../repos')
+
+async function getCompany (req, res, next, id) {
```
Comment (ztolley) [inline src/apps/companies/middleware/params.js]: There's already a middleware named getCompanyDetails that is doing this and extracting the company name and address for the heading but it is only designed to work for non ch and only being used for interactions. We will need to remove that from the router (line 77) and replace it with something that is called after this to calculate the heading title and address only.

**Comment by tyom** [comment]
[Companies House URL](http://localhost:3000/companies/view/ch/05588682) seems to be broken.

**Comment by teneightfive** [comment]
>Companies House URL seems to be broken.

I did spot that too. It's being addressed in a [follow up PR](https://github.com/uktrade/data-hub-frontend/pull/679) to refactor companies view routes.

File: src/apps/companies/router.js [hunk 2]
```diff
@@ -20,11 +20,16 @@ const {
...
 const { getRequestBody, getCompanyCollection } = require('./middleware/collection')
 const { populateForm, handleFormPost } = require('./middleware/form')
+const { getCompany, getCompaniesHouseRecord } = require('./middleware/params')
 const { setDefaultQuery } = require('../middleware')
 
 const DEFAULT_COLLECTION_QUERY = {
   sortby: 'modified_on:desc',
 }
+
+router.param('id', getCompany)
```
Comment (tyom) [inline src/apps/companies/router.js]: `getCompanyRecord` to match the line below?

File: src/apps/companies/transformers.js [hunk 3]
```diff
@@ -103,7 +105,43 @@ function transformCompaniesHouseCompanyToListItem (props) {
   })
 }
 
+function transformCompanyFromApi (body) {
```
Comment (tyom) [inline src/apps/companies/transformers.js]: I don't think this belongs here in this form.

**Pull request merged**